packages/core/__tests__/utils/generateRandomString.test.ts

```diff
@@ -3,28 +3,48 @@
 
 import { generateRandomString } from '../../src/libraryUtils';
 
-describe('generateRandomString()', () => {
-	it('generates a string with the specified length', () => {
-		let counter = 0;
-		while (counter++ < 50) {
-			expect(generateRandomString(20).length).toEqual(20);
-		}
+describe('generateRandomString', () => {
+	const getRandomValuesSpy = jest.spyOn(crypto, 'getRandomValues');
+	beforeAll(() => {
+		// Mock crypto.getRandomValues
+		getRandomValuesSpy.mockImplementation(bufferView => {
+			if (!bufferView) {
+				return null;
+			}
+			const array = new Uint8Array(bufferView.buffer);
+			for (let i = 0; i < array.byteLength; i++) {
+				array[i] = Math.floor(Math.random() * 256);
+			}
+
+			return array;
+		});
 	});
 
-	it('generates correct string', () => {
-		const mathRandomSpy = jest.spyOn(Math, 'random');
-		let counter = 1;
-		mathRandomSpy.mockImplementation(() => {
-			const returnValue = counter;
-			counter += 5;
+	afterEach(() => {
+		getRandomValuesSpy.mockClear();
+	});
 
-			return parseFloat(`0.${returnValue}`);
-		});
+	it('generates a string of the specified length', () => {
+		const length = 10;
+		const result = generateRandomString(length);
+		expect(result).toHaveLength(length);
```

Comment:
Nit: I think it's probably ok to just be concise here
```suggestion
		expect(generateRandomString(length)).toHaveLength(10);
```